lib/js/stanford_events.behavior.js

```diff
@@ -21,6 +21,40 @@ export default {
           $(this).attr('aria-expanded', 'false');
         }
       });
+
+      // Get today's date
+      var today = new Date();
+
+      // Update so today is all of the day
+      var a = today.setHours(0, 0, 0, 0);
+      var todayBegin = parseInt(a, 10);
+
+      // Location for label
+      const eventType =
+        $('div.node-stanford-event-su-event-type .su-event-type');
+
+      // Location for text
+      const eventTextLocation =
+        $('div.node-stanford-event-su-event-date-time');
+
+      // Looking for the date on the node page
+      const eventDate =
+        $('section.event').attr('data-end-date');
+
+      // Build the labels and sentence
+      const pastLabel =
+        $('<span class="su-event-label-past">Past Event ' +
+          '<span class="divider">|</span></span>');
+
+      const pastText =
+        $('<div class="su-event-text-past">This event has passed.</div>');
+
+      // Apply past label and text
+      if (eventDate < todayBegin) {
```

Comment:
You could do this check earlier before settings the various variables. it will make a minor performance improvement.

Reply:
Ok.